pkg/operator/types/types.go

```diff
@@ -103,6 +128,61 @@ func (r Resource) GetNamespace() string {
 	return ""
 }
 
+func (r Resource) ShouldWaitForReady() bool {
+	if r.Unstructured != nil {
+		annotations := r.Unstructured.GetAnnotations()
+		if annotations == nil {
+			return false
+		}
+		waitForReady, ok := annotations[WaitForReadyAnnotation]
+		if !ok {
+			return false
+		}
+		return waitForReady == "true"
+	}
+	return false
+}
+
+func (r Resource) ShouldWaitForProperties() bool {
+	if r.Unstructured != nil {
+		annotations := r.Unstructured.GetAnnotations()
+		if annotations == nil {
+			return false
+		}
+		_, ok := annotations[WaitForPropertiesAnnotation]
+		return ok
+	}
+	return false
+}
+
+// GetWaitForProperties returns the key value pairs in the `kots.io/wait-for-properties` annotation
+func (r Resource) GetWaitForProperties() []WaitForProperty {
+	if r.Unstructured != nil {
+		annotations := r.Unstructured.GetAnnotations()
+		if annotations == nil {
+			return nil
+		}
+		annotationValue, ok := annotations[WaitForPropertiesAnnotation]
+		if !ok {
+			return nil
+		}
+
+		waitForProperties := []WaitForProperty{}
+		for _, property := range strings.Split(annotationValue, ",") {
+			parts := strings.SplitN(property, "=", 2)
+			if len(parts) != 2 {
+				continue
+			}
```

Comment:
Should we log here so we know if something was skipped because it was formatted incorrectly?